src/components/generics/vacation-test-forms/employee-vacation-requests.tsx

```diff
@@ -92,6 +102,72 @@ const RenderEmployeeVacationRequests = ({ persons }: Props) => {
     initializeRequests();
   }, [person]);
 
+  /**
+   * Initializes all vacation request statuses
+   */
+  const initializeRequestStatuses = async () => {
+    try {
+      const vacationRequestStatuses: VacationRequestStatus[] = [];
+      const statusesApi = Api.getVacationRequestStatusApi(accessToken?.access_token);
+
+      await Promise.all(requests.map(async request => {
+        let createdStatuses: VacationRequestStatus[];
+        if (request.id) {
+          createdStatuses = await statusesApi.listVacationRequestStatuses({ id: request.id });
+        } else {
+          throw new Error("Vacation Request ID undefined!");
+        }
+        createdStatuses.forEach(createdStatus => {
+          vacationRequestStatuses.push(createdStatus);
+        });
+      }));
+
+      setStatuses(vacationRequestStatuses);
+      setLoading(false);
+    } catch (error) {
+      context.setError(strings.errorHandling.fetchVacationDataFailed, error);
+    }
+  };
+
+  useEffect(() => {
+    if (!requests.length) {
+      return;
+    }
+    initializeRequestStatuses();
+  }, [requests]);
+
+  /**
+   * Initializes all the latest vacation request statuses, so there would be only one status for each request showed on the UI
+   */
+  const initializeLatestStatuses = async () => {
+    const latestStatuses: VacationRequestStatus[] = [];
+
+    requests.forEach(request => {
+      const requestStatuses: VacationRequestStatus[] = [];
+      
+      statuses.forEach(pickedStatus => {
+        if (request.id === pickedStatus.vacationRequestId) {
+          requestStatuses.push(pickedStatus);
+        }
+      });
+  
+      // Pick the latest statuses
+      if (requestStatuses.length > 0) {
+        const pickedStatus = requestStatuses.reduce((a, b) => (a.updatedAt! > b.updatedAt! ? a : b));
+        latestStatuses.push(pickedStatus);
+      }
+    });
+
+    setLatestStatuses(latestStatuses);
+  };
+
+  useEffect(() => {
+    if (statuses.length <= 0) {
```

Comment:
The previous comment referred to shorting the statuses.length condition- it seems like some of the suggests were positioned incorrectly last time...

Reply:
This one might as well be shortened as well
```suggestion
      return;
```